index.php

```diff
@@ -270,6 +274,18 @@ function get_q_value($mime, $accept)
         error_page('Login Failed', 'Invalid username or password.');
     }
 
+    $scope = filter_input_regexp(INPUT_POST, 'scopes', '@^[\x21\x23-\x5B\x5D-\x7E]+$@', FILTER_REQUIRE_ARRAY);
```

Comment:
I had an unexpected output trying to use an emoji as scope. Suggesting this does not return `false` when one of the scopes is wrong, but will update the `scopes` array to contain `false`. This means the following check against `false` does nothing and the `implode()` will actually add extra spaces.

Needs some more testing before merging!